pkg/kubelet/nodelease/controller_test.go

```
@@ -282,3 +282,111 @@ func TestRetryUpdateLease(t *testing.T) {
+			updateReactor: nil,
+			getReactor: func(action clienttesting.Action) (bool, runtime.Object, error) {
+				return true, nil, notFoundErr
+			},
+			createReactor: func(action clienttesting.Action) (b bool, object runtime.Object, e error) {
+				return true, &coordinationv1.Lease{ObjectMeta: metav1.ObjectMeta{ResourceVersion: "1"}}, nil
+			},
+			onRepeatedHeartbeatFailure: nil,
+			expectLeaseResourceVersion: "1",
+		},
+		{
+			desc:        "latestLease is nil and need to update",
+			latestLease: nil,
+			updateReactor: func(action clienttesting.Action) (bool, runtime.Object, error) {
+				return true, &coordinationv1.Lease{ObjectMeta: metav1.ObjectMeta{ResourceVersion: "2"}}, nil
+			},
+			getReactor: func(action clienttesting.Action) (bool, runtime.Object, error) {
+				return true, &coordinationv1.Lease{ObjectMeta: metav1.ObjectMeta{ResourceVersion: "1"}}, nil
+			},
+			onRepeatedHeartbeatFailure: nil,
+			expectLeaseResourceVersion: "2",
+		},
+		{
+			desc:        "latestLease exist and need to update",
+			latestLease: &coordinationv1.Lease{ObjectMeta: metav1.ObjectMeta{ResourceVersion: "1"}},
+			updateReactor: func(action clienttesting.Action) (bool, runtime.Object, error) {
+				return true, &coordinationv1.Lease{ObjectMeta: metav1.ObjectMeta{ResourceVersion: "2"}}, nil
+			},
+			onRepeatedHeartbeatFailure: nil,
+			expectLeaseResourceVersion: "2",
+		},
+		{
+			desc:        "update with latest lease failed",
+			latestLease: &coordinationv1.Lease{ObjectMeta: metav1.ObjectMeta{ResourceVersion: "1"}},
+			updateReactor: func() func(action clienttesting.Action) (b bool, object runtime.Object, e error) {
+				i := 0
+				return func(action clienttesting.Action) (bool, runtime.Object, error) {
+					i++
+					switch i {
+					case 1:
+						return true, nil, notFoundErr
+					case 2:
+						return true, &coordinationv1.Lease{ObjectMeta: metav1.ObjectMeta{ResourceVersion: "3"}}, nil
+					default:
+						return true, &coordinationv1.Lease{}, nil
```

this should never happen - so instead of failing silently, it's better to do t.Fatalf(....)

---

done